qiskit_nature/operators/second_quantization/star_algebra_mixin.py

```diff
@@ -0,0 +1,99 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Sum Operator base interface."""
+
+from typing import Union
+from abc import ABC, abstractmethod
+
+
+class StarAlgebraMixin(ABC):
+    """The star algebra mixin class.
+
+    Star algebra means algebra with adjoint (dagger).
+    """
+
+    @abstractmethod
+    def mul(self, other: complex):
```

Comment:
I think this splitting of the actual logic and the operator overloads is a good idea. However, in this specific case `complex` is quite restrictive. We should check what the opflow in Terra uses as the type here.

Reply:
`float` and `int` is a virtual subclass of `complex` in mypy. So I think this code is valid.